test/array_serializer_test.rb

```diff
@@ -39,15 +45,52 @@ def test_meta_and_meta_key_attr_readers
         assert_equal @serializer.meta_key, "the meta key"
       end
 
-      def test_json_key_when_resource_is_empty
```

Comment:
Ah, all good. The test name might've been what was confusing. Maybe it makes more sense to say `test_json_key_with_empty_resource_with_name` instead of `test_json_key_with_resource_with_name_and_no_serializers`. Though it looks like you already merged.